src/Nest/QueryDsl/Query.cs

```diff
@@ -114,11 +110,11 @@ public static class Query<T> where T : class
 		public static QueryContainer ParentId(Func<ParentIdQueryDescriptor<T>, IParentIdQuery> selector) =>
 			new QueryContainerDescriptor<T>().ParentId(selector);
 
-		public static QueryContainer Percolate(Func<PercolateQueryDescriptor<T>, IPercolateQuery> selector) =>
-			new QueryContainerDescriptor<T>().Percolate(selector);
-
-		public static QueryContainer Prefix(Expression<Func<T, object>> fieldDescriptor, string value, double? boost = null, MultiTermQueryRewrite rewrite = null, string name = null) =>
-			new QueryContainerDescriptor<T>().Prefix(fieldDescriptor, value, boost, rewrite, name);
+		public static QueryContainer Percolate(Func<PercolateQueryDescriptor<T>, IPercolateQuery> selector) =>
+			new QueryContainerDescriptor<T>().Percolate(selector);
+
+		public static QueryContainer Prefix(Expression<Func<T, object>> fieldDescriptor, string value, double? boost = null, MultiTermQueryRewrite rewrite = null, string name = null) =>

```

Comment:
Wondering if we should remove all the shorthands that accept `Field`, `Expression<Func<T,object>>` et. al, except for on `Term()` and `Terms()`?

Reply:
Or atleast get rid of the nullable parameters. I think we should do this for `7.x` at this stage though